server/zio-http-server/src/test/scala/sttp/tapir/server/ziohttp/ZioHttpTestServerInterpreter.scala

```diff
@@ -21,15 +23,14 @@ class ZioHttpTestServerInterpreter(eventLoopGroup: EventLoopGroup, channelFactor
 
   override def server(routes: NonEmptyList[Http[Any, Throwable, Request, Response]]): Resource[IO, Port] = {
     implicit val r: Runtime[Any] = Runtime.default
-    val layers: ZLayer[Any, Nothing, EventLoopGroup with ServerChannelFactory] =
-      ZLayer.succeed(eventLoopGroup) ++ ZLayer.succeed(channelFactory)
-
-    val server: Server[Any, Throwable] = Server.app(routes.toList.reduce(_ ++ _))
 
-    val io: ZIO[Scope, Throwable, Server.Start] = ZIO
-      .scoped(Server.make(server ++ Server.port(0)))
-      .provide(layers)
+    val io: ZIO[Scope, Throwable, Int] =
+      (for {
+        driver <- ZIO.service[Driver]
+        _ <- driver.start(trace)
+        _ <- Server.serve(routes.toList.reduce(_ ++ _)).provide(ZLayer.succeed(driver) >>> Server.base).forkScoped
+      } yield 0).provide(Scope.default, ServerConfig.live(ServerConfig(address = new InetSocketAddress(0))), NettyDriver.default)
```

Comment:
`provideSome[Scope]` perhaps since it looks like you want to preserve the scope into the `Resource`

Reply:
That's the same as doing Scope.default, no? 🙂

Reply:
```
  /**
   * A layer that constructs a scope and closes it when the workflow the layer
   * is provided to completes execution, whether by success, failure, or
   * interruption. This can be used to close a scope when providing a layer to a
   * workflow.
   */
```

So this will close the scope when the workflow completes,, but the type ascription is still showing `ZIO[Scope, Throwable, Int]` so it seems like you want the scope to actually be tied to the `Resource.scoped[IO, Any, Int](io)` no?

Reply:
Ah, very true, had to look it up. I essentially thought the two were equivalent (more or less `ZIO.service[Scope]`) – but they're really not. I'll change it and see if it makes a difference!